mypy/main.py

```diff
@@ -102,120 +104,152 @@ def process_options(args: List[str]) -> Tuple[List[BuildSource], Options]:
             module to run as script (or None),
             parsed flags)
     """
-    # TODO: Rewrite using argparse.
+
+    footer = textwrap.dedent(
+        """environment variables:
+        MYPYPATH     additional module search path"""
+    )
+
+    parser = argparse.ArgumentParser(prog='mypy', epilog=footer,
+                                     formatter_class=argparse.RawDescriptionHelpFormatter)
+
+    def parse_version(v):
+        version_components = v.split(".")[0:2]
+        if len(version_components) != 2:
+            raise argparse.ArgumentTypeError(
+                "Invalid python version '{}' (expected format: 'x.y')".format(v))
+        if not all(item.isdigit() for item in version_components):
+            raise argparse.ArgumentTypeError("Found non-digit in python version: '{}'".format(v))
+        return (int(version_components[0]), int(version_components[1]))
+
+    parser.add_argument('-v', '--verbose', action='store_true', help="more verbose messages")
+    parser.add_argument('--version', action='version', version='%(prog)s ' + __version__)
+    parser.add_argument('--python-version', type=parse_version, metavar='x.y',
+                        help='use Python x.y')
+    parser.add_argument('--py2', action='store_true', help="use Python 2 mode")
```

Comment:
Yep, that's better! I forgot to go back and clean it up after realizing that's the only thing that `--py2` did.  Changed.